src/web/routes/application/middleware/handle-path-request/handle-path-request.js

```diff
@@ -1,20 +1,21 @@
-const { CHECK_URL, TERMS_AND_CONDITIONS_URL, CONFIRM_URL } = require('../common/constants')
-const { isGuidancePageUrl } = require('../../guidance/predicates')
-const { stateMachine, actions, states } = require('../common/state-machine')
+const { CONFIRM_URL } = require('../../common/constants')
+const { stateMachine, actions, states } = require('../../common/state-machine')
+const { isPathInSequence, stepNotNavigable } = require('./predicates')
+const { getPathsInSequence } = require('./selectors')
 
 const { IS_PATH_ALLOWED, GET_NEXT_ALLOWED_PATH } = actions
 
-const getPathsInSequence = (steps) => [...steps.map(step => step.path), CHECK_URL, TERMS_AND_CONDITIONS_URL, CONFIRM_URL]
-
-const stepNotNavigable = (step, req) => step && typeof step.isNavigable === 'function' && !step.isNavigable(req.session)
-
 const middleware = (config, pathsInSequence, step) => (req, res, next) => {
   // Destroy the session on navigating away from CONFIRM_URL
   if (stateMachine.getState(req) === states.COMPLETED && req.path !== CONFIRM_URL) {
     req.session.destroy()
     res.clearCookie('lang')
-    const redirectPath = isGuidancePageUrl(req.path) ? req.path : config.environment.OVERVIEW_URL
-    return res.redirect(redirectPath)
+
+    if (isPathInSequence(req.path, pathsInSequence)) {
+      return res.redirect(config.environment.OVERVIEW_URL)
+    }
+
+    return next()
   }
```

Comment:
Lines 14 -> 18 are the important parts of this PR:

- Remove redirect by default - call `next()` if path is allowed
- Assert if requested path is **in** sequence instead of asserting for paths **outside** of the sequence